### ISSUE TITLE
Fix parsing of invalid JS object, handling forward slash instead of comma

### ISSUE THREAD

**Issue by abebus** [Nykakin/chompjs#60]
I'm currently extracting the following data from a <script> tag on the website:

```
var a = {
  'compositions': [
    'Shell: Cotton 100%' /
    'Pocket lining:Cotton 100%'
  ]
}
```
However, it appears that this is not a valid JavaScript object. The _chompjs library handles this data like this:

```
_chompjs.parse(a)
'{"compositions":["Shell: Cotton 100%""/\n    \'Pocket lining:Cotton 100%\'\n  "]}'
```
However, when I try to parse this data using the chompjs.parse_js_object function, I get the following error:

```
chompjs.parse_js_object(a)
Traceback (most recent call last):
...
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 38 (char 37)
```
To fix this issue, I have found that replacing the forward slashes with commas in the data fixes the parsing error:

```
chompjs.parse_js_object(a.replace('/', ','))
{'compositions': ['Shell: Cotton 100%', 'Pocket lining:Cotton 100%']}

```

**Comment by Nykakin** [comment]
Not sure what this slash is supposed to be. JavaScript evaluates this to:
```javascript
{ compositions: [ NaN ] }
```
And it indeed messes with the parsing.

Can you post URL to the website in question?

**Comment by abebus** [comment]
Yes, sure. https://www.cos.com/en_eur/women/womenswear/dresses/product.open-back-tiered-midi-dress-yellow.1238217002.html

**Comment by Nykakin** [comment]
This seems to be website-specific so I don't really want to modify parser just for this. At the same time I don't really like fixing with `str.replace` because if other data contains `/` inside, it will be replaced too. At the very least you should use something more precise, such as
```python
import re

chompjs.parse_js_object(re.sub(r"/\w*\n", ',', a))
```
Ideally you'd want an option to allow customizable separators, but this is not easy to have.

Not sure yet how to handle this properly. Leaving opened for now as I'll have to think it through.

**Comment by Nykakin** [comment]
Closing as inactive.
